src/backend/web/graphql/index.js

```diff
@@ -94,7 +122,27 @@ module.exports.resolvers = {
         const last = first + prPage < numOFPosts ? first + prPage : numOFPosts;
         const postIds = await getPosts(first, last);
 
-        return Promise.all(postIds.map(id => Post.byId(id)));
+        const result = await Promise.all(postIds.map(id => Post.byId(id)));
+
+        if (filter) {
+          // check if author name is equal to what we're searching for
+          if (filter.author) {
+            return result.filter(post => post.author === filter.author);
+          }
+          // check if post's date is between two provided dates
+          if (filter.fromDate || filter.toDate) {
+            const fromDate = filter.fromDate ? filter.fromDate : new Date();
+            const toDate = filter.toDate ? filter.toDate : new Date();
+            const datePostIds = await getPostsByDate(fromDate, toDate);
```

Comment:
I think you need to check if you're doing a date-filter earlier, since there is no point getting all posts on line 123 if you are going to do it again here by date.  Move your filter checks up, and decide which way to get posts based on that.

Reply:
I think this was the question I wanted to ask, are we going to do the pagination first. As in

1. I want to see 30 posts on page 0
2. Out of those 30 posts I want to see the ones that are from "February 2, 2020" to "February 07, 2020". I'll then compare which of those posts on page 0 are within the results from `getPostsByDate()`

Reply:
Pagination only makes sense in the context of your data, so calculate the posts you have first, then figure out pages after that